### DISCUSSION TITLE
Is Tailwind CSS users violating the MIT license?

### DISCUSSION THREAD

**Discussion by unsungNovelty** [tailwindlabs/tailwindcss#11094] [General]
Hi, quick question regarding the MIT license compliance.

Are we, the users in violation of MIT license since I don't see the copyright notice of the license itself in the output file?
All I see is `/*! tailwindcss v3.0.24 | MIT License | https://tailwindcss.com*/*`.
To be clear I am not talking about the github repo of Tailwind CSS. But the output .css file generated with Tailwind CSS.

From my understanding of the MIT license, the end user should include the copyright notice in it's entirety. From MIT license text:
> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

Which is the only condition in the MIT license. Which means when I am using Tailwind CSS, the generated output css file should include the below:

```
MIT License

Copyright (c) Tailwind Labs, Inc.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
```
---

*Reading around, I came across the below which confirms my doubt - https://drewdevault.com/2022/02/07/Free-software-licenses-MIT.html:*
> All of this is subject to the conditions of paragraph two, of which there is only one: you must include the copyright notice and license text in any substantial copies or derivatives of the software. Thus, the MIT license requires attribution. This can be achieved by simply including the full license text (copyright notice included) somewhere in your project. For a proprietary product, this is commonly hidden away in a menu somewhere. For a free software project, where the source code is distributed alongside the product, I often include it as a comment in the relevant files. You can also add your name or the name of your organization to the list of copyright holders when contributing to MIT-licensed projects, at least in the absence of a CLA.

---

If I am indeed right about the MIT license violation, changing the output file which already says MIT license: `/*! tailwindcss v3.0.24 | MIT License | https://tailwindcss.com*/*` to include the copyright notice should solve the issue. Otherwise, everyone is violating the MIT license unless they specifically include the copyright notice after generating the output .css file. Which is always error prone and cumbersome. 

PS: Do let me know if I should be opening a ticket as well. I thought it would be a good idea to open up a discussion before creating a ticket.

**Comment by arhandev11** [comment]
is there any answer for this question? i kinda curious about this too

**Comment by TheXenocide** [comment]
It may be helpful to allow some sort of template text for output files to be configured as I would really like to be able to put a comment at the beginning of the file that indicates that the file is generated (to warn users not to edit it directly)

**Comment by mitar** [comment]
I am not a lawyer, but I am not sure if CSS output should even have a MIT license? Isn't this like compiled output from a compiler? The license of the compiler does not apply to the compiled output?

I think the fact that TailwindCSS adds that banner is misleading and they should not be doing that.

And if you say that in the case of TailwindCSS it combines pre-made (licensed) parts together, then I say that compilers do that as well. A compiler has a pre-made part for how to convert a for loop in C into assembly, for example.

**Comment by mitar** [comment]
There is a followup to [the discussion here](https://github.com/tailwindlabs/tailwindcss/issues/11536#issuecomment-1618990213).
